src/frontends/tensorflow_lite/src/op/while.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <openvino/opsets/opset11.hpp>
+
+#include "common_op_table.hpp"
+#include "op_translation_utils.hpp"
+#include "utils.hpp"
+
+using namespace std;
+using namespace ov::opset11;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow_lite {
+namespace op {
+
+OutputVector while_op(const ov::frontend::tensorflow_lite::NodeContext& node) {
+    const auto& decoder = get_decoder(node);
+    int32_t cond_idx = decoder->get_attribute(&tflite::WhileOptions::cond_subgraph_index);
+    int32_t body_idx = decoder->get_attribute(&tflite::WhileOptions::body_subgraph_index);
+
```

Comment:
the code is quite similar to TF While translator. Can we create `translate_session` class for storing cached previously converted body graphs. I think it makes sense to do because in this case we can use `Frontend` class inplementation for `convert` method that contain logic for telemetry.

I am leaning to align this code with TF implementation.

Reply:
I have considered it. However, upon the closer look into the TF FE translation session I have encountered lots of logic which is TF specific -- there is no need for it from TFL FE perspective. Remaining code is quite simple. This is why I kept translation logic in InputModel and overloaded [public frontend::NodeContext API|https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/common/include/openvino/frontend/node_context.hpp#L101:L111] that we have right now.

With all that `frontend::tensorflow_lite::NodeContext` is able to trigger translation on demand.

With regards to the alignment -- It would be great to pull all the similar code into common space. Perhaps passing condition / body as `ov::Model` would be a good start. None the less, there are several places where TF implementation has shape / type setting for better TensorList and DT_VARIANT support. These parts are not needed for TFL.

I had an attempt to unify this code, but it broke into several pieces which made it less elegant. So, I left them separate for now. I believe we will return to this unification question when unification along all the frontends would come. But for now I don't see the urgency in it.

Reply:
Yes, there are some TF specific stuff in `TranslateSession` but you can easily skip them and design the common `TranslateSession` class and inherit TFL `TranslateSession` class with your specific methods.

My point is to have the common TranslateSession with common `get_converted_model` in the future. Now you can overload this function if we don't have FW node support in TFL FE.

Now if we decline use of `TranlateSession`, we create additional gap from moving to the common Frontend where we can place telemetry stuff, for example.